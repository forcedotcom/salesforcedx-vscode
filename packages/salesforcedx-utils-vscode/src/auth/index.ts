/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

/**
 * This file is intentionally left blank. AuthUtil is exported
 * in the root of salesforcedx-utils-vscode.
 */
