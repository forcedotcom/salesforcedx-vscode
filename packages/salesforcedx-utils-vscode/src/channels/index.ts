/*
 * Copyright (c) 2017, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import { ChannelService } from './channelService';
export const channelService = ChannelService.getInstance();
export { DEFAULT_SFDX_CHANNEL } from './channelService';
export { ChannelService } from './channelService';
