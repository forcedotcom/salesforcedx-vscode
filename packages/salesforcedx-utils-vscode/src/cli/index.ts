/*
 * Copyright (c) 2017, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

export interface Command {
  readonly command: string;
  readonly description?: string;
  readonly args: string[];
  readonly logName?: string;

  toString(): string;
  toCommand(): string;
}

export { CommandBuilder, SfdxCommandBuilder } from './commandBuilder';
export {
  CliCommandExecutor,
  CliCommandExecution,
  CommandExecution,
  CompositeCliCommandExecutor,
  GlobalCliEnvironment
} from './commandExecutor';
export { CommandOutput } from './commandOutput';
export { ForceConfigGet } from './forceConfigGet';
export { ForceOrgDisplay, OrgInfo } from './forceOrgDisplay';
export { LocalCommandExecution } from './localCommandExecutor';
export {
  ForceDeployResultParser,
  ForceSourceDeployErrorResponse,
  ForceSourceDeploySuccessResponse,
  DeployResult,
  CONFLICT_ERROR_NAME
} from './deployResultParser';
export { TestRunner } from './testRunner';
export {
  OrgCreateSuccessResult,
  OrgCreateErrorResult,
  OrgCreateResultParser
} from './orgCreateResultParser';
export {
  OrgOpenSuccessResult,
  OrgOpenContainerResultParser,
  OrgOpenErrorResult
} from './orgOpenContainerResultParser';
export {
  DiffErrorResponse,
  DiffResultParser,
  DiffSuccessResponse
} from './diffResultParser';
