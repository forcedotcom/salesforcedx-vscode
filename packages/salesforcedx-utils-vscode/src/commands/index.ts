/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

export { ChannelService } from './channelService';

import { NotificationService } from './notificationService';
export const notificationService = NotificationService.getInstance();
export { NotificationService };
export { ProgressNotification } from './progressNotification';
