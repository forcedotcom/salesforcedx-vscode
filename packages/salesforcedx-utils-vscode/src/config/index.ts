/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

/**
 * This file is intentionally left blank. ConfigSource and ConfigUtil are exported
 * in the root of salesforcedx-utils-vscode.
 */
