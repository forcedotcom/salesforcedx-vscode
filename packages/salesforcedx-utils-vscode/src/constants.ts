export const SETTING_CLEAR_OUTPUT_TAB = 'clearOutputTab';
export const SFDX_CORE_CONFIGURATION_NAME = 'salesforcedx-vscode-core';
export const SFDX_CORE_EXTENSION_NAME = 'salesforcedx-vscode-core';
