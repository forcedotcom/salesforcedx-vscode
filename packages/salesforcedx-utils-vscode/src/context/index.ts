/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

/**
 * This file is intentionally left blank. WorkspaceContextUtil and OrgInfo are exported
 * in the root of salesforcedx-utils-vscode.
 */
