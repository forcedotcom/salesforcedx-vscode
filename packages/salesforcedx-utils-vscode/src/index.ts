/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

export * from './commands';
export {
  CompositeParametersGatherer,
  EmptyParametersGatherer
} from './commands/parameterGatherers';
export { EmptyPostChecker } from './commands/postconditionCheckers';
export {
  EmptyPreChecker,
  SfdxWorkspaceChecker
} from './commands/preconditionCheckers';
export {
  SfdxCommandletExecutor,
  LibraryCommandletExecutor
} from './commands/commandletExecutors';
export { SfdxCommandlet } from './commands/sfdxCommandlet';
export { ConfigSource, ConfigUtil } from './config/configUtil';
export {
  SETTING_CLEAR_OUTPUT_TAB,
  SFDX_CORE_CONFIGURATION_NAME,
  SFDX_CORE_EXTENSION_NAME
} from './constants';
export {
  OrgUserInfo,
  WorkspaceContextUtil
} from './context/workspaceContextUtil';
export {
  TelemetryService,
  TelemetryBuilder,
  TelemetryData,
  Properties,
  Measurements
} from './telemetry/telemetry';
export * from './helpers';
export { TraceFlags } from './helpers/traceFlags';
export { TelemetryReporter } from './telemetry/telemetryReporter';
export {
  hasRootWorkspace,
  getRootWorkspace,
  getRootWorkspacePath,
  workspaceUtils
} from './workspaces';
export * from './cli';
export * from './cli/commandExecutor';
export * from './i18n';
export * from './types';
export * from './date';
export * from './output';
export * from './predicates';
export * from './providers';
export * from './services';
