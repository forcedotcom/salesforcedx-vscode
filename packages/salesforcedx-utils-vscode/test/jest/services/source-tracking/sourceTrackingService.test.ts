/*
 * Copyright (c) 2023, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
import { SourceTracking } from '@salesforce/source-tracking';
import { WorkspaceContextUtil } from '../../../../src';
import { SourceTrackingService } from '../../../../src/services';
import { testData } from './testData';

jest.mock('@salesforce/core', () => ({
  ...jest.requireActual('@salesforce/core'),
  Org: { create: jest.fn() },
  SfProject: { resolve: jest.fn() }
}));

describe('Source Tracking Service', () => {
  describe('createSourceTracking', () => {
    let sourceTrackingCreateSpy: jest.SpyInstance;
    let ensureLocalTrackingSpy: jest.SpyInstance;

    beforeEach(() => {
      sourceTrackingCreateSpy = jest
        .spyOn(SourceTracking, 'create')
        .mockResolvedValue({} as any);
      ensureLocalTrackingSpy = jest
        .spyOn(SourceTracking.prototype, 'ensureLocalTracking')
        .mockResolvedValue({} as any);
    });

    it('Should create an instance of SourceTracking', async () => {
      await SourceTrackingService.createSourceTracking('', {} as any);

      expect(sourceTrackingCreateSpy).toHaveBeenCalled();
    });
  });

  describe('updateSourceTrackingAfterRetrieve', () => {
    const updateTrackingFromRetrieveSpy = jest.fn();
    const dummySourceTracking = {
      updateTrackingFromRetrieve: updateTrackingFromRetrieveSpy
    } as any;

    it('Should update an instance of SourceTracking using the retrieve result', async () => {
      const dummyRetrieveResult = {} as any;
      await SourceTrackingService.updateSourceTrackingAfterRetrieve(
        dummySourceTracking,
        dummyRetrieveResult
      );

      expect(updateTrackingFromRetrieveSpy).toHaveBeenCalledWith(
        dummyRetrieveResult
      );
    });
  });

  describe('getSourceStatusSummary', () => {
    const mockWorkspaceContextUtil = {
      onOrgChange: jest.fn(),
      getConnection: jest.fn()
    };
    const updateTrackingFromRetrieveSpy = jest.fn();
    const getStatusMock = jest.fn();
    const dummySourceTracking = {
      updateTrackingFromRetrieve: updateTrackingFromRetrieveSpy
    } as any;

    let workspaceContextUtilGetInstanceSpy: jest.SpyInstance;
    let sourceTrackingMock: jest.SpyInstance;

    beforeEach(() => {
      workspaceContextUtilGetInstanceSpy = jest
        .spyOn(WorkspaceContextUtil, 'getInstance')
        .mockReturnValue(mockWorkspaceContextUtil as any);

      sourceTrackingMock = jest
        .spyOn(SourceTracking, 'create')
        .mockResolvedValue({
          getStatus: getStatusMock
        } as any);
    });

    it('Should return a properly formatted string when changes exist in the response', async () => {
      // Arrange
      getStatusMock.mockResolvedValue(testData.statusResponse as any);

      // Act
      const formattedOutput: string = await SourceTrackingService.getSourceStatusSummary(
        {}
      );

      // Assert
      expect(workspaceContextUtilGetInstanceSpy).toHaveBeenCalled();
      expect(sourceTrackingMock).toHaveBeenCalled();
      expect(formattedOutput).toMatchSnapshot();
    });

    it('Should return a friendly message when no changes exist', async () => {
      getStatusMock.mockResolvedValue(testData.noChangesResponse as any);

      const formattedOutput: string = await SourceTrackingService.getSourceStatusSummary(
        {}
      );

      expect(workspaceContextUtilGetInstanceSpy).toHaveBeenCalled();
      expect(sourceTrackingMock).toHaveBeenCalled();
      expect(formattedOutput).toMatchSnapshot();
    });
  });
});
