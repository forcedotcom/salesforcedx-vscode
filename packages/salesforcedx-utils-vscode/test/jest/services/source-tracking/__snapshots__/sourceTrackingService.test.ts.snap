// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source Tracking Service getSourceStatusSummary Should return a friendly message when no changes exist 1`] = `"No local or remote changes found."`;

exports[`Source Tracking Service getSourceStatusSummary Should return a properly formatted string when changes exist in the response 1`] = `
"IGNORED  STATE          FULL NAME                         TYPE       PROJECT PATH                                                       
───────  ─────────────  ────────────────────────────────  ─────────  ───────────────────────────────────────────────────────────────────
false    Local Changed  CommunitiesLandingController      ApexClass  force-app/main/default/classes/CommunitiesLandingController.cls    
false    Local Changed  CommunitiesLandingControllerTest  ApexClass  force-app/main/default/classes/CommunitiesLandingControllerTest.cls
false    Local Changed  ProductController                 ApexClass  force-app/main/default/classes/ProductController.cls               
false    Remote Add     Default                           Audience                                                                      
true     Remote Add     Admin                             Profile                                                                       
"
`;
